docs/developer/extending/webhooks/synchronous-events/circuit-breakers.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Sync webhooks circuit breakers
+description: Documentation for synchronous webhooks circuit breakers in Saleor
+---
+
+Circuit breakers are a mechanism that temporarily shuts down an app in case of a high number of webhook errors. The feature is off by default.
+
+## Breaker states
+
+### Closed
+
+This is a normal operation state, all webhooks are sent as usual.
+
+### Open
+
+The breaker is open when the number of errors exceeds the threshold. In this state, all webhooks are rejected.
+
+### Half-open
+
+In this state, the breaker allows the requests like in closed state (allowing requests to pass through), however using different thresholds.
+:::warning
+
+This is different from common implementation of breakers in which the requests are throttled in half-open state.
+
+:::
+
+## Transitions between states
+
+:::info
+
+Breaker states are being changed automatically as the requests go by. The breaker is monitoring the events and act accordingly.
+
+:::
+
+### Closed to Open
+
+The conditions on which breaker enters open state are controlled by two constants:
+- `BREAKER_BOARD_FAILURE_MIN_COUNT` - it defaults to 100 and is the minimum number of failures that must occur before the breaker can trip.
+- `BREAKER_BOARD_FAILURE_THRESHOLD_PERCENTAGE` - it defaults to 35 (percent) and is the ratio of failures to total requests that must be exceeded before the breaker trips, assuming `BREAKER_BOARD_FAILURE_MIN_COUNT` is satisfied.
+
+### Open to Half-open
+
+After the breaker enters open state, no webhooks are sent for a certain amount of time. This time is called the cooldown period and it defaults to 2 minutes. Then the breaker automatically transitions to the half-open state.
+
+
+### Half-open back to Open
+
+The breaker will trip again if the number of errors exceeds the threshold and minimum count. The conditions on which breaker re-enters open state are controlled by another two constants:
+- `BREAKER_BOARD_FAILURE_MIN_COUNT_RECOVERY` - simmiliar to `BREAKER_BOARD_FAILURE_MIN_COUNT`, but it defaults to 20 and is the minimum number of failures that must occur before the breaker can trip again.
+- `BREAKER_BOARD_FAILURE_THRESHOLD_PERCENTAGE_RECOVERY` - simmiliar to `BREAKER_BOARD_FAILURE_THRESHOLD_PERCENTAGE`, but it defaults to 30 (percent) and is the ratio of failures to total requests that must be exceeded before the breaker trips again.
+
+## Half-open back to Closed
+
+If the number of successful requests is met, the breaker transitions back to the closed state. This is controlled by `BREAKER_BOARD_SUCCESS_COUNT_RECOVERY` constant, which defaults to 50.
+
+Note that all of circuit breaker events (both success and failures) have a certain TTL (time to live) and are stored in Redis. On each transition, the events are purged, so each state has a fresh start and is not contaminated by older events.
+
+### Configuring the feature
```

Comment:
And this as headining 2?

Reply:
And maybe it's worth to move it before state transition, but that's just cosmetic.